src/stripe/objects/core/event.cr

```diff
@@ -8,8 +8,11 @@ class Stripe::Event
     getter idempotency_key : String?
   end
 
-  alias StripeObject = Stripe::Product | Stripe::Invoice | Stripe::Subscription | Stripe::Price | Stripe::Coupon | Stripe::Plan
+  alias StripeObject = Stripe::Customer | Stripe::Product | Stripe::Invoice | Stripe::Subscription | Stripe::Price | Stripe::Coupon | Stripe::Plan | Stripe::Balance | Stripe::Charge | Stripe::Payout | Stripe::PaymentIntent | Stripe::SetupIntent | Stripe::PromotionCode | Stripe::Source | Stripe::TaxRate # | Stripe::Checkout::Session
```

Comment:
I wonder if there's a way to generate this with a macro that would search for all `Stripe` subclasses? This seems like it could be really easy to forget to update as the library expands.

I'm not overly familiar with macros, though, so I'm not 100% sure if that's even possible.

Reply:
Ok, after more live tests, this "magic solution" is not working as expected :
- The JSON parsing can generate erroneous Stripe objects (a `Stripe::Coupon` instead of a `Stripe::Product` for example).

There might be a JSON::Builder way of doing things, but it might be overkill.

IMO, a safer solution is to make this `getter object` a `JSON::Any` and let the user use the proper `Stripe::Object.from_json` on it. (or add a method which detects the `object : String` value and builds the corresponding and correct object). If this last sentence is confusing is because the Stripe api docs are misleading in saying :

> Object containing the API resource relevant to the event. For example, an invoice.created event will have a full invoice object as the value of the object key.

There is a sub-object key inside the object in the sent json with the name of the object :

```json
  "data": {
    "object": {
      "object": "balance",
```

Also, the `Stripe::Event` `previous_attributes` getter is missing as it is optional (not always present in the sent event payloads).

I welcome any guidance before going further :-)

Reply:
@confact you're a bit better at this macro stuff than I am... any ideas/suggestions on where to take this?

Reply:
@stephendolan sorry, I missed your message. I will try to put some time to try something out. We would need to create a static macro with case checks to return the class based on the object string. Ugly but could work.

I need to try first, though, to be sure if it would work.